pkgs/os-specific/linux/plymouth/default.nix

```diff
@@ -1,102 +1,112 @@
 { lib
 , stdenv
-, fetchpatch
 , fetchFromGitLab
+, meson
 , pkg-config
-, autoreconfHook
+, libpng
+, pango
+, gtk3
+, libdrm
+, libevdev
+, libxkbcommon
+, xorg
 , libxslt
+, ninja
 , docbook-xsl-nons
-, gettext
-, gtk3
 , systemd
-, pango
-, cairo
-, libdrm
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "plymouth";
-  version = "unstable-2021-10-18";
+  version = "unstable-2023-06-05";
 
-  outputs = [
-    "out"
-    "dev"
-  ];
+  outputs = [ "out" "dev" ];
 
   src = fetchFromGitLab {
     domain = "gitlab.freedesktop.org";
     owner = "plymouth";
     repo = "plymouth";
-    rev = "18363cd887dbfe7e82a2f4cc1a49ef9513919142";
-    sha256 = "sha256-+AP4ALOFdYFt/8MDXjMaHptkogCwK1iXKuza1zfMaws=";
+    rev = "a5eda165689864cc9a25ec14fd8c6da458598f42";
+    hash = "sha256-TpMZZ0naC4D0Knmclc8JpmXPfnpM6q8YotIkNX+aRVo=";
   };
 
+  patches = [
+    ./dont-create-broken-symlink.patch
+    ./add-runtime-plugin-path.patch
+  ];
+
   nativeBuildInputs = [
-    autoreconfHook
-    docbook-xsl-nons
-    gettext
-    libxslt
+    meson
     pkg-config
+    ninja
+    docbook-xsl-nons
   ];
 
   buildInputs = [
-    cairo
+    libpng
+    pango
     gtk3
     libdrm
-    pango
+    libevdev
+    libxkbcommon
+    xorg.xkeyboardconfig
+    libxslt
     systemd
   ];
 
-  postPatch = ''
-    sed -i \
-      -e "s#plymouthplugindir=.*#plymouthplugindir=/etc/plymouth/plugins/#" \
-      -e "s#plymouththemedir=.*#plymouththemedir=/etc/plymouth/themes#" \
-      -e "s#plymouthpolicydir=.*#plymouthpolicydir=/etc/plymouth/#" \
-      -e "s#plymouthconfdir=.*#plymouthconfdir=/etc/plymouth/#" \
-      configure.ac
-  '';
-
-  configurePlatforms = [ "host" ];
-
-  configureFlags = [
-    "--enable-documentation"
-    "--enable-drm"
-    "--enable-gtk"
-    "--enable-pango"
-    "--enable-systemd-integration"
-    "--enable-tracing"
-    "--localstatedir=/var"
+  mesonFlags = [
     "--sysconfdir=/etc"
-    "--with-background-color=0x000000"
-    "--with-background-end-color-stop=0x000000"
-    "--with-background-start-color-stop=0x000000"
-    "--with-logo=/etc/plymouth/logo.png"
-    "--with-release-file=/etc/os-release"
-    "--with-runtimedir=/run"
-    "--with-systemdunitdir=${placeholder "out"}/etc/systemd/system"
-    "--without-rhgb-compat-link"
-    "--without-system-root-install"
-    "ac_cv_path_SYSTEMD_ASK_PASSWORD_AGENT=${lib.getBin systemd}/bin/systemd-tty-ask-password-agent"
+    "--localstatedir=/var"
+    "-Dlogo=/etc/plymouth/logo.png"
+    "-Dbackground-color=0x000000"
+    "-Dbackground-start-color-stop=0x000000"
+    "-Dbackground-end-color-stop=0x000000"
+    "-Drelease-file=/etc/os-release"
+    "-Dtracing=true"
+    "-Dsystemd-integration=true"
```

Comment:
Many of these are enabled by default with Meson port: https://gitlab.freedesktop.org/plymouth/plymouth/-/blob/a5eda165689864cc9a25ec14fd8c6da458598f42/meson_options.txt